Inc/HALAL/Services/Communication/UART/UART.hpp

```diff
@@ -108,35 +112,84 @@ class UART{
      */
     static void start();
 
-    /**@brief	Transmits 1 RawPacket of any size by DMA and
-     *          interrupts. Handles the packet size automatically. To
-     *          to send various packets in a row you must check if the UART is busy
-     *          using is_busy().
+    /**@brief	Transmits 1 byte by DMA and interrupts.
+     *          To send various packets in a row you must check if the UART is busy
+     *          using is_busy(). All calls to this functions previous
+     *          the bus is ready be ignored.
      * 
      * @param id Id of the UART
-     * @param packet Packet to be send
+     * @param data data to be send
      * @return bool Returns true if the request to send the packet has been done
      *            successfully. Returns false if the UART is busy or a problem
      *            has occurred.
      */
-    static bool transmit_next_packet(uint8_t id, RawPacket& packet);
+    static bool transmit(uint8_t id, uint8_t data);
+
+    /**@brief	Transmits size number of bytes by DMA and interrupts.
+	 *          To send various packets in a row you must check if the UART is busy
+	 *          using is_busy(). All calls to this functions previous
+     *          the bus is ready be ignored.
+	 *
+	 * @param id Id of the UART
+	 * @param data Data to be sent.
+	 * @param size Size of the data
+	 * @return bool Returns true if the request to send the packet has been done
+	 *            successfully. Returns false if the UART is busy or a problem
+	 *            has occurred.
+	 */
+    static bool transmit(uint8_t id, uint8_t* data, int16_t size);
```

Comment:
```suggestion
    static bool transmit(uint8_t id, array data);
```
Si vas a cambiarlo en el SPI cambialo tambien en el UART. Es hacer data.size() y data.data(), no hace falta que cambies nada de la funcion.